src/Mailbox.php

```diff
@@ -41,6 +42,28 @@ public function getName(): string
         return $this->name;
     }
 
+    public function renameTo(string $name = null): bool
+    {
+        if (null === $name) {
+            return false;
+        }
+        $encodedName = \mb_convert_encoding($name, 'UTF7-IMAP', 'UTF-8');
+        $oldFullName = $this->getFullEncodedName();
+        $newFullName = \preg_replace('/' . \preg_quote(\mb_convert_encoding($this->name, 'UTF7-IMAP', 'UTF-8')) . '$/', $encodedName, $oldFullName);
+        if (null === $newFullName) {
+            return false;
+        }
+
+        $return = \imap_renamemailbox($this->resource->getStream(), $oldFullName, $newFullName);
+        if (false === $return) {
+            throw new RenameMailboxException('Could not rename mailbox');
```

Comment:
We need to test this: just try to rename the mailbox to itself, it should suffice